### ISSUE TITLE
Use covariant returns for `Owner` property in accessible objects

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#7974]
Since C# 9 we can use [covariant returns](https://learn.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/covariant-returns), though we never got around enhancing the code. Updating the code will allow us to remove the duplicate fields and use `Owner` property throughout.

_Originally posted by @RussKie in https://github.com/dotnet/winforms/pull/7954#discussion_r998992435_

**Comment by Tanya-Solyanik** [comment]
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md#source-and-binary-compatibility-changes:

```
Adding virtual to a member

While this change would often work without breaking too many scenarios because C# compiler tends to emit callvirt IL instructions to call non-virtual methods (callvirt performs a null check, while a normal call won't), we can't rely on it. C# is not the only language we target and the C# compiler increasingly tries to optimize callvirt to a normal call whenever the target method is non-virtual and the this is provably not null (such as a method accessed through the ?. null propagation operator). Making a method virtual would mean that consumer code would often end up calling it non-virtually.
```

**Comment by merriemcgaw** [comment]
We need to start with a new internal property on `ControlAccessibleObject` and then as we work on others we pick them up.

**Comment by NikitaSemenovAkvelon** [comment]
Tried to use covariant return for new property `OwnerInternal`, but got a `TypeLoadException` in `Marshal.GetIUnknownForObject` method:

> System.TypeLoadException
  HResult=0x80131522
  Message=Signature of the body and declaration in a method implementation do not match.  Type: 'ButtonBaseAccessibleObject'.  Assembly: 'System.Windows.Forms, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Runtime.InteropServices.Marshal.GetIUnknownForObjectNative(Object o)
   at System.Windows.Forms.Control.WmGetObject(Message& m) in C:\Projects\winforms.main\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 12152

https://github.com/dotnet/winforms/blob/1b8ac24a2b13f78eae47e177abfe8a59658372de/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs#L12152
@AaronRobinsonMSFT do you have any thoughts on why Marshal throws the exception? Or maybe I do something wrong?

**Comment by AaronRobinsonMSFT** [comment]
@NikitaSemenovAkvelon I don't know for sure, but this is likely some validation done in the COM interop subsystem. The gist being, does the marshallable object adhere to some interface contract. The notion of covariant returns isn't something that COM interop is likely aware of. You should be able to create a very small repro case for this and if it continues to fail, please file an issue in dotnet/runtime.

The general idea is below.

```csharp
using System.Runtime.InteropServices;'

[ComImport]
[Guid("7d2a86ba-826c-4167-aaf3-483777f4ba10")]
public interface IFoo
{
    A Prop { get; }    
}

[ComVisible(true)]
public class Foo : IFoo
{
    B Prop { get; }   
}
```

**Comment by RussKie** [comment]
@AaronRobinsonMSFT the puzzling thing is that we're trying to _add_ a new internal covariant property:

```cs
using System.Runtime.InteropServices;'

[ComImport]
[Guid("7d2a86ba-826c-4167-aaf3-483777f4ba10")]
public interface IFoo
{
    A Prop { get; }    
}

[ComVisible(true)]
public class Foo : IFoo
{
    public B Prop { get; }   
    internal B PropInternal => Prop;
}

public class FooBar : Foo
{
    internal C PropInternal => Prop; // C is a descendant of B
}
```

**Comment by NikitaSemenovAkvelon** [comment]
Oh, sorry I've forgotten to attach the branch with a commit.
https://github.com/NikitaSemenovAkvelon/winforms/tree/Issue_7974-Refactoring_ControlAccessibleObject_OwnerCovariantReturn

**Comment by NikitaSemenovAkvelon** [comment]
Created an Issue about this bug https://github.com/dotnet/runtime/issues/78090

**Comment by NikitaSemenovAkvelon** [comment]
Looks like the exception won't be fixed (https://github.com/dotnet/runtime/issues/78090#issuecomment-1309457613). Thereby we can't resolve this issue.
